src/Type/Union/MenuItemObjectUnionType.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * MenuItemObjectUnionType
+ */
+
+namespace WPGraphQL\Type\Union;
+
+use GraphQL\Type\Definition\UnionType;
+use WPGraphQL\Types;
+
+/**
+ * Class MenuItemObjectUnionType
+ *
+ * Navigation menus comprise menu items that reference an object, which can be
+ * a post object, a taxonomy term object, or a custom link.
+ */
+class MenuItemObjectUnionType extends UnionType {
```

Comment:
We should add a WPUnionType to the core plugin to help centralize filters, etc. 🤔